activestorage/lib/active_storage/attached/model.rb

```diff
@@ -153,17 +153,7 @@ def #{name}=(attachables)
           end
         CODE
 
-        has_many :"#{name}_attachments", -> { where(name: name) }, as: :record, class_name: "ActiveStorage::Attachment", inverse_of: :record, dependent: :destroy, strict_loading: strict_loading do
-          def purge
-            each(&:purge)
-            reset
-          end
-
-          def purge_later
-            each(&:purge_later)
-            reset
-          end
```

Comment:
I think we can remove this:
- This extension is no longer called from the `Attached` API
- I didn't see any documentation explicitly referring to `#{name}_attachments.purge/purge_later`
- No tests are using this extension explicitly

For the above reasons shouldn't be a breaking change and allows us to avoid a maintenance burden.

Reply:
@intrip Are these public API? We could add a deprecation (in a follow-up PR)

Reply:
Honestly I'm not sure (hence my comment above): Is public API a method which has `public` visibility but is not documented and doesn't have an explicitly `# :nodoc:` comment? 🤷‍♂️